### PR TITLE
Added a .add() function

### PR DESCRIPTION
**Pull request by seanballais**
Since I read that @RodRitter wants to a .add() function, so I coded up the function. It still needs to be criticized though.

### CODE REVIEW

File: kew.js [hunk 1]
```diff
@@ -38,6 +38,40 @@
 			for (var i = 0; i < this.length; i++) {
 				this.elements[i].parentElement.removeChild(this.elements[i]);
 			}
+		},
+		add: function(htmlString) {
+			if (htmlString == "") {
+				throw new Error("Parameter in .add must not be an empty string.");
+			}
+
+			for (var i = 0; i < this.length; i++) {
```
Comment (thomastuts) [inline kew.js]: You can use the `this.each()` method here instead of writing a loop yourself.

Comment (seanballais) [inline kew.js]: @thomastuts, it would make the code inconsistent if I use `this.each()`.

Comment (thomastuts) [inline kew.js]: By using the reusing the same variable `i` inside your nested loop, you are essentially creating a **memory leak**.

Try using `kew.js` in a browser window - if you use the `add()` method, your tab gets frozen and keeps executing code until you close it. My tab went up to 5.8GB memory used before I had to kill it using Chrome's task manager.

Comment (seanballais) [inline kew.js]: Oh, I see. Thanks! Never thought of that. Any explanations why it'll be creating a **memory leak** if I kept reusing the same variable?

**Comment by thomastuts** [comment]
> Oh, I see. Thanks! Never thought of that. Any explanations why it'll be creating a memory leak if I kept reusing the same variable?

To be honest, I'm not completely sure yet myself. Changing `i` to `j` in the for loop (or using `this.each()` for the first one like you've done now) definitely fixes it though. Pretty sure it has something to do with adding the elements to the array we're iterating on, resulting in an infinite loop. I'll take a look at it later and see if I can come up with a feasible explanation.

> it would make the code inconsistent if I use this.each().

I don't really see how it should, seeing as you're using a built-in method? Isn't it best to reuse that logic in case the `.each()` method gets optimized somehow in the future, instead of basically repeating the logic?

Also, I'm not sure why I bolded memory leak in my comment. It seemed appropriately dramatic. :sparkles:

**Comment by seanballais** [comment]
Makes sense. Thanks! :)

**Comment by PAEz** [comment]
I dont understand why you think `i` is resulting in memory leaks?...but meh....
The reason you had an infinite loop was you have a for loop in a for loop and their both using `i` as their indexing variable....thats gonna cause problems ;)  You dont wanna be doing that till we have `let`.
And whats with this add function anywayz?  I dont or every have used jquery or any framework like it, so I could just be missing something.
Whats with the slack parsing looking for a tag and all that?
Couldnt you just something like.....

``` javascript
var tmp = document.createElement('body'), child;
tmp.innerHTML = '<div>x</div><span>y</span>';
while (child = tmp.firstElementChild) {
    elemQueried.appendChild(child);
}
```

If only IE had `template`

**Comment by seanballais** [comment]
@PAEz Thanks for insight. The add() function is a feature @RodRitter wanted to add so I attempted to implement it.

And what is the value, and data type of `child`?

**Comment by PAEz** [comment]
`And what is the value, and data type of child?`
What ever the value and type of tmp.firstElementChild is ;P
Im not entirely sure what add is meant to do, but what the above does is covert html to dom and attach it to something.
Just noticed that it wouldnt add text that was in the html, this should fix that...

``` javascript
var elemQueried = document.body;
var tmp = document.createElement('body'), child;
tmp.innerHTML = '<div>x</div>Bleh<span>y</span>';
var children = tmp.childNodes;
while(children.length){
    elemQueried.appendChild(children[0]);
}
```

I got that from....
http://stackoverflow.com/questions/9284117/inserting-arbitrary-html-into-a-documentfragment
...answered by the great Rob W

**Comment by RodRitter** [comment]
Thanks for contributing Sean! I was looking into this and I think we are going to need `prepend()` and `append()` functions instead of a generic `add()` - I think what you are trying to do is add an element to the end of the element . Does this make sense?

**Comment by seanballais** [comment]
Yeah something like that.

**Pull request closed without merging**